src/status_im2/contexts/onboarding/sign_in/view.cljs

```diff
@@ -1,42 +1,226 @@
+      (rf/dispatch [:syncing/input-connection-string-for-bootstrapping data])
+      (rf/dispatch [:toasts/upsert
+                    {:icon       :info
+                     :icon-color colors/danger-50
+                     :text       (i18n/label :t/error-this-is-not-a-sync-qr-code)}]))))
 
-(defn sign-in
+(defn view
   []
-  [rn/view {:style {:flex 1}}
-   [background/view true]
-   [page]])
+  (let [active-tab                 (reagent/atom 1)
+        camera-permission-granted? (reagent/atom false)
+        qr-view-finder             (reagent/atom {})
+        height                     (rf/sub [:dimensions/window-height])
+        width                      (rf/sub [:dimensions/window-width])]
+    [:f>
+     (fn []
+       (let [insets                    (safe-area/use-safe-area)
+             show-holes?               (and @qr-view-finder
+                                            (= @active-tab 1)
+                                            @camera-permission-granted?)
+             holes                     (merge @qr-view-finder {:borderRadius 16})
+             show-camera?              (and (= @active-tab 1) @camera-permission-granted?)
+             camera-ref                (atom nil)
+             read-qr-once?             (atom false)
+             on-read-code              (fn [data]
+                                         (when-not @read-qr-once?
+                                           (reset! read-qr-once? true)
+                                           (check-qr-code-data data)))
+             request-camera-permission (fn []
+                                         (rf/dispatch
+                                          [:request-permissions
+                                           {:permissions [:camera]
+                                            :on-allowed  #(reset! camera-permission-granted? true)
+                                            ;; TODO: Check with the design team for unhappy flow
+                                            :on-denied   #(rf/dispatch
+                                                           [:toasts/upsert
+                                                            {:icon       :info
+                                                             :icon-color colors/danger-50
+                                                             :text       "Permission denied"}])}]))]
+         (rn/use-effect
+          (fn []
+            (permissions/permission-granted? :camera
+                                             #(reset! camera-permission-granted? %)
+                                             #(reset! camera-permission-granted? false))))
+         [rn/view {:style {:flex 1 :padding-top (:top insets)}}
+          (if show-camera?
+            [camera
+             {:ref            #(reset! camera-ref %)
+              :style          (merge style/absolute-fill {:height height :width width})
+              :camera-options {:zoomMode :off}
```

Comment:
Is it possible to use kebab case here?

Reply:
No, it doesn't work if we use a kebab case.